### PR TITLE
Make `RangeTool` respect `Range.{min,max}_interval`

### LINKED ISSUE #13483: [BUG] RangeTool does not respect target `max_interval`
### Software versions

Python version      :  3.10.12 | packaged by conda-forge | (main, Jun 23 2023, 22:41:52) [Clang 15.0.7 ]
IPython version     :  8.16.1
Tornado version     :  6.3.3
Bokeh version       :  3.3.0
BokehJS static path :  /Users/droumis/opt/miniconda3/envs/bokeh3.3/lib/python3.10/site-packages/bokeh/server/static
node.js version     :  v20.4.0
npm version         :  9.7.2
Operating system    :  macOS-13.5.2-arm64-arm-64bit

### Browser name and version

_No response_

### Jupyter notebook / Jupyter Lab version

_No response_

### Expected behavior

The RangeTool should respect the constraints imposed by the `max_interval` on the target plot and not allow for expanding the range beyond this interval. Depending on the intended behavior, this may be somewhere between a bug and a feature request.

### Observed behavior

The RangeTool ignores the `max_interval` constraint.

### Example code

```Python
import numpy as np
from bokeh.layouts import column
from bokeh.models import ColumnDataSource, RangeTool
from bokeh.plotting import figure, show

n_points = 3000
x_values = np.linspace(0, 100, n_points) 
y_values = np.random.randn(n_points).cumsum()

source = ColumnDataSource(data=dict(x=x_values, y=y_values))

p = figure(height=300, width=800, tools=["xpan", "xzoom_in", "xzoom_out", "reset"],
           background_fill_color="#efefef", x_range=(22, 30))
p.line('x', 'y', source=source)
p.yaxis.axis_label = 'Value'
p.x_range.max_interval = 10  # Set max zoom interval

select = figure(height=130, width=800, y_range=p.y_range,
                tools="", toolbar_location=None, background_fill_color="#efefef")
range_tool = RangeTool(x_range=p.x_range)
range_tool.overlay.fill_color = "navy"
range_tool.overlay.fill_alpha = 0.2
select.line('x', 'y', source=source)
select.ygrid.grid_line_color = None
select.add_tools(range_tool)

show(column(p, select))
```

### Stack traceback or browser console output

_No response_

### Screenshots

https://github.com/bokeh/bokeh/assets/6613202/fd17e84b-4694-4458-9cae-c0e3f756a745

### PR DESCRIPTION
**Pull request by mattpap**
Makes `RangeTool` respect its `x_range` and `y_range` limits. Also adds support for `{min,max}_{width,height}` constraints to `BoxAnnotation`. These constraints work, but overall the support is cumbersome in edge cases and will require additional work in a future PR, to polish the behavior. This PR also adds `View.on_transitive_change` for connecting `change` signals of transitive models in a view. I originally added this in PR #13538, but it's useful here as well and harder to miss in a much smaller PR. This approach is generic enough to later be used in all other context where previously we either had custom code or no signal reconnect/cleanup at all.

fixes #13483

### CODE REVIEW

File: bokehjs/src/lib/core/view.ts [hunk 1]
```diff
@@ -129,6 +129,29 @@ export class View implements ISignalable {
     }
   }
 
+  on_transitive_change<T>(property: Property<T>, collect: (value: T) => HasProps[], fn: () => void): void {
```
Comment (philippjfr) [inline bokehjs/src/lib/core/view.ts]: This watches for changes on child models?

**Review by philippjfr — APPROVED** [review]
Looks good to me!

File: bokehjs/src/lib/core/view.ts (hunk 1 above)
Comment (mattpap) [reply bokehjs/src/lib/core/view.ts]: It watches for models provided by `collect()`, which allows working with a subset of child models (for this property) and doesn't require generalized traversal of data structures.

Comment (mattpap) [reply bokehjs/src/lib/core/view.ts]: Actually, I dropped `collect()` in favour of `HasProps.references()`, which greatly simplifies usage.

**Pull request merged**